samza-kafka/src/main/scala/org/apache/samza/system/kafka/KafkaSystemFactory.scala

```diff
@@ -97,7 +97,8 @@ class KafkaSystemFactory extends SystemFactory with Logging {
       systemName,
       new ExponentialSleepStrategy(initialDelayMs = producerConfig.reconnectIntervalMs),
       getProducer,
-      metrics)
+      metrics,
+      dropProducerExceptions = config.getDropProducerError)
```

Comment:
Minor: can remove the parameter name.

Reply:
IIUC then the config would be used as the clock parameter for KafkaSystemProducer. Doesn't compile when I tried it.